pkg/controller/accountclaim/organizational_units.go

```diff
@@ -124,12 +122,10 @@ func MoveAccount(reqLogger logr.Logger, client awsclient.Client, account *awsv1a
 					return awsv1alpha1.ErrAccAlreadyInOU
 				}
 			case "ConcurrentModificationException":
-				// if we encounter a race condition we simply try again after a random sleep interval
-				r := rand.Intn(5)
-				ConcurrentModificationExceptionMsg := fmt.Sprintf("CreateOrganizationalUnit:ConcurrentModificationException: Race condition while attempting to move Account: %s to OU: %s, trying again in %d seconds", account.Spec.AwsAccountID, OUID, r)
+				// if we encounter a race condition we simply log the condition and return
+				ConcurrentModificationExceptionMsg := fmt.Sprintf("OU:CreateOrganizationalUnit:ConcurrentModificationException: Race condition while attempting to move Account: %s to OU: %s", account.Spec.AwsAccountID, OUID)
 				reqLogger.Info(ConcurrentModificationExceptionMsg)
-				time.Sleep(time.Duration(r) * time.Second)
-				MoveAccount(reqLogger, client, account, OUID, parentID)
+				return nil
```

Comment:
why return nil and not an error?

Reply:
Because this error indicates that the thing we wanted to do is already done. Which could be made clearer in the comment and/or log message.

However, that's only the case because we happen to know this is the only thing we're doing to this resource. It would be safer and more future-proof to requeue (either by returning the error or return with `Requeue: true`) and let the subsequent reconcile make sure the thing is in the right state.

Reply:
Got it, guess I missed that this error still means the $stuff got done. Something more explicit would be nice here